csharp/src/Ice/LocatorDiscovery/Locator.cs

```diff
@@ -246,27 +169,31 @@ private Locator(Communicator communicator)
             communicator.SetProperty("Ice.LocatorDiscovery.Reply.ProxyOptions", "-d");
             communicator.SetProperty("Ice.LocatorDiscovery.Reply.AcceptNonSecure", "Always");
 
+            _instanceName = communicator.GetProperty("Ice.LocatorDiscovery.InstanceName") ?? "";
+
             _replyAdapter = communicator.CreateObjectAdapter("Ice.LocatorDiscovery.Reply");
             _locatorAdapter = communicator.CreateObjectAdapter();
 
+            var locatorIdentity = new Identity("Locator",
+                                               _instanceName.Length > 0 ? _instanceName : Guid.NewGuid().ToString());
+            Proxy = _locatorAdapter.Add(locatorIdentity, this, ILocatorPrx.Factory);
+
```

Comment:
```suggestion
            _instanceName = communicator.GetProperty("Ice.LocatorDiscovery.InstanceName") ?? Guid.NewGuid().ToString();

            _replyAdapter = communicator.CreateObjectAdapter("Ice.LocatorDiscovery.Reply");
            _locatorAdapter = communicator.CreateObjectAdapter();

            var locatorIdentity = new Identity("Locator", _instanceName );
            Proxy = _locatorAdapter.Add(locatorIdentity, this, ILocatorPrx.Factory);

```

Reply:
Yes, can't put a UUID in instance name, as an "empty" instance name is assigned by the first discovered locator.